Content.Shared/Ninja/Components/NinjaSuitComponent.cs

```diff
@@ -0,0 +1,166 @@
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Ninja.Systems;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Ninja.Components;
+
+/// <summary>
+/// Component for ninja suit abilities and power consumption.
+/// As an implementation detail, dashing with katana is a suit action which isn't ideal.
+/// </summary>
+[Access(typeof(SharedNinjaSuitSystem))]
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class NinjaSuitComponent : Component
+{
+    [ViewVariables, AutoNetworkedField]
+    public bool Cloaked = false;
+
+    /// <summary>
+    /// The action for toggling suit phase cloak ability
+    /// </summary>
+    [DataField("togglePhaseCloakAction")]
+    public InstantAction TogglePhaseCloakAction = new()
+    {
+        UseDelay = TimeSpan.FromSeconds(5), // have to plan un/cloaking ahead of time
+        DisplayName = "action-name-toggle-phase-cloak",
+        Description = "action-desc-toggle-phase-cloak",
+        Priority = -9,
+        Event = new TogglePhaseCloakEvent()
+    };
```

Comment:
Stealth closing functionality should be split out from ninja star + emp, each one should just be generic clothing actions that drain power to use (using PowerCellDraw).

Reply:
made pr